core/optaplanner-core-impl/src/main/java/org/optaplanner/core/config/ruin/RuinPhaseConfig.java

```diff
@@ -0,0 +1,77 @@
+package org.optaplanner.core.config.ruin;
+
+import java.util.function.Consumer;
+
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlType;
+
+import org.optaplanner.core.config.heuristic.selector.move.generic.*;
+import org.optaplanner.core.config.phase.PhaseConfig;
+import org.optaplanner.core.config.solver.termination.TerminationConfig;
+import org.optaplanner.core.config.util.ConfigUtils;
+
+@XmlType(propOrder = {
+        "moveSelectorConfig"
+})
+public class RuinPhaseConfig extends PhaseConfig<RuinPhaseConfig> {
+
+    public static final String XML_ELEMENT_NAME = "ruin";
+
+    @XmlElement(name = RuinMoveSelectorConfig.XML_ELEMENT_NAME, type = RuinMoveSelectorConfig.class,
+            namespace = "https://www.optaplanner.org/xsd/solver")
+    private RuinMoveSelectorConfig moveSelectorConfig = null;
+
+    // ************************************************************************
+    // Constructors and simple getters/setters
+    // ************************************************************************
+
+    public RuinMoveSelectorConfig getMoveSelectorConfig() {
+        return moveSelectorConfig;
+    }
+
+    public void setMoveSelectorConfig(RuinMoveSelectorConfig moveSelectorConfig) {
+        this.moveSelectorConfig = moveSelectorConfig;
+    }
+
+    public TerminationConfig getTerminationConfig() {
+        TerminationConfig terminationConfig = super.getTerminationConfig();
+        if (terminationConfig != null) {
+            return terminationConfig;
+        }
+        TerminationConfig defaultTerminationConfig = new TerminationConfig();
+        defaultTerminationConfig.setStepCountLimit(1);
+        return defaultTerminationConfig;
+    }
```

Comment:
The ruin phase needs no termination. 
It does one move, and the move needs to finish.
I suggest overriding `setTermination(...)` to throw `UnsupportedOperationException`.
`getTerminationConfig()` should probably return `null` or whatever is convenient.